functions.php

```diff
@@ -0,0 +1,36 @@
+<?php
+
+function createAllDisplayCards(array $results): string {
+    $outputString = '';
+
+    foreach ($results as $card) {
+        if (!is_array($card) ||
+            !array_key_exists('title', $card) ||
+            !array_key_exists('cardType', $card) ||
+            !array_key_exists('color', $card) ||
+            !array_key_exists('raritySet', $card)) {
+            return '';
+        }
+        $cardString = "<div class='display-card'>";
+        $cardString .= "<p>Card Title: ". $card['title'] . "</p>";
+        $cardString .= "<p>Card Type: " . $card['cardType'] . "</p>";
+        $cardString .= "<p>Card Color: " . $card['color'] . "</p>";
+        $cardString .= "<p>Rarity: " . $card['raritySet'] . "</p>";
+        $cardString .= "</div>";
+        $outputString .=  $cardString;
+    }
+ return $outputString;
```

Comment:
Indent this to line up with previous line.

Reply:
Indenting done.